app/services/google_oauth.rb

```diff
@@ -56,10 +56,22 @@ def user_email(credentials)
       service.get_userinfo.email
     end
 
+    def push_notification_client
+      Google::Apis::CalendarV3::CalendarService.new.tap { |s| s.authorization = authorization_details }
+    end
+
+    def authorization_details
+      scopes = [CALENDAR_SCOPE]
+      authorization = Google::Auth.get_application_default(scopes)
+      authorization.sub = ENV.fetch('APPLICATION_OWNER')
```

Comment:
And this `authorization` objects looks like an interesting candidate for `#tap`.